TFNetworkLayer.py

```diff
@@ -2981,15 +2980,16 @@ def _get_out_type_from_opts(cls, n_out, filter_size, padding, strides=1, dilatio
     padding = padding.upper()
     if input_expand_dims == 0 and not input_add_feature_dim and not input_split_feature_dim:
       # Maybe we have a chance to correctly define the output shapes.
-      index_shift = data.time_dim_axis_excluding_batch
+      if data.is_batch_feature_major:
+        data = data.copy_with_feature_dim_axis(-1)
```

Comment:
Why? Remove that. Use `index_shift` as you did before (for `data.shape`; not for `shape`, as I wrote earlier). There is no need in an additional copy of the input (would be totally wasted resources).